feat: add media entity to ResolveRemoteThumbnailUrlExtension (#11557)
shopware/platform: changelog/_unreleased/2025-07-31-add-mediaEntity-to-resolveremotethumbnailurlextension.md

---

> title: add mediaEntity to ResolveRemoteThumbnailUrlExtension

Reviewer: ```suggestion
title: Add MediaEntity to ResolveRemoteThumbnailUrlExtension
```

author: Sascha Heilmeier
author_email: sascha.heilmeier@netlogix.de
author_github: @scarbous
---
# Core

> * Use `Shopware\Core\Content\Media\MediaEntity` in `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow individual media handling.

Reviewer: ```suggestion
* Added `Shopware\Core\Content\Media\MediaEntity` to `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow individual media handling.
```

> * add `mimeType` to `Shopware\Core\Content\Media\Core\Params\UrlParams` 

Reviewer: ```suggestion
* Added `mimeType` to `Shopware\Core\Content\Media\Core\Params\UrlParams` 
```

> * allow to skipp thumbnail by returning `null` as result through `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension`

Reviewer: ```suggestion
* Changed `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow to skip thumbnail loading by returning `null` as result
```

> * Changed `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow to skip thumbnail loading by returning `null` as result

Reviewer: sorry, I was mainly concentrating on the code :see_no_evil: but as we introduced some deprecations/breaks for the next major version, we need to extend the changelog. 

add a section `# Upgrade Information` where you describe what was now deprecated and what the alternative is

after that the section `# Next Major Version Changes` which will be part of the 6.8 upgrade guide and which tells, what was removed and what to use instead

Author: Thank you for your support 🙏🏼

I hope I get it right 😊

Author: @mitelg can you review my changes :)

---
title: Add MediaEntity to ResolveRemoteThumbnailUrlExtension
author: Sascha Heilmeier
author_email: sascha.heilmeier@netlogix.de
author_github: @scarbous
---
# Core
* Added `Shopware\Core\Content\Media\MediaEntity` to `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow individual media handling.
* Added `mimeType` to `Shopware\Core\Content\Media\Core\Params\UrlParams` 
* Changed `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow to skip thumbnail loading by returning `null` as result
___
# Upgrade Information
___
* The properties `$mediaPath` and `$mediaUpdatedAt` from `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension`, they could be resolved from `$mediaEntity`
# Next Major Version Changes
## The properties `$mediaPath` and `$mediaUpdatedAt` from `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` get removed

> * Use `$mediaEntity` instead of the `$mediaPath` and `$mediaUpdatedAt` properties.

Reviewer: ```suggestion

___

# Upgrade Information

## Deprecation of properties in `ResolveRemoteThumbnailUrlExtension`

The properties `mediaPath` and `mediaUpdatedAt` from `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` are deprecated and will be removed with the next major version. Set the values directly into the newly added `mediaEntity` property.

___

# Next Major Version Changes

## Removal of properties in `ResolveRemoteThumbnailUrlExtension`

The properties `$mediaPath` and `$mediaUpdatedAt` from `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` were removed. Set the values directly into the `mediaEntity` property.
```

---
title: Add MediaEntity to ResolveRemoteThumbnailUrlExtension
author: Sascha Heilmeier
author_email: sascha.heilmeier@netlogix.de
author_github: @scarbous
---
# Core
* Added `Shopware\Core\Content\Media\MediaEntity` to `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow individual media handling.
* Added `mimeType` to `Shopware\Core\Content\Media\Core\Params\UrlParams` 
* Changed `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` to allow to skip thumbnail loading by returning `null` as result

___

# Upgrade Information

## Deprecation of properties in `ResolveRemoteThumbnailUrlExtension`

The properties `mediaPath` and `mediaUpdatedAt` from `Shopware\Core\Content\Media\Extension\ResolveRemoteThumbnailUrlExtension` are deprecated and will be removed with the next major version. Set the values directly into the newly added `mediaEntity` property.

## Deprecation of `media` and `thumbnail` in `MediaPathChangedEvent`

> The method `media` from `Shopware\Core\Content\Media\Event\MediaPathChangedEvent` are deprecated and will be removed with the next major version. Use the newly added `mediaWithMimeType` method instead.
> 
> The method `thumbnail` from `Shopware\Core\Content\Media\Event\MediaPathChangedEvent` are deprecated and will be removed with the next major version. Use the newly added `thumbnailWithMimeType` method instead.

Reviewer 2: ```suggestion
The method `media` from `Shopware\Core\Content\Media\Event\MediaPathChangedEvent` is deprecated and will be removed with the next major version. Use the newly added `mediaWithMimeType` method instead.

The method `thumbnail` from `Shopware\Core\Content\Media\Event\MediaPathChangedEvent` is deprecated and will be removed with the next major version. Use the newly added `thumbnailWithMimeType` method instead.
```
